core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -80,6 +89,7 @@ public Configuration conf() {
   @Override
   public void initialize(Map<String, String> props) {
     this.properties = SerializableMap.copyOf(props);
+    storageCredentialConfig().forEach(getConf()::set);
```

Comment:
do you have an actual use case for this? It's unclear to me what kind of storage credentials a REST server would send to a client that would apply for `HadoopFileIO`

Reply:
> do you have an actual use case for this? It's unclear to me what kind of storage credentials a REST server would send to a client that would apply for `HadoopFileIO`

The credentials being passed are used to access an OSS bucket that belongs to the REST service. This bucket is not publicly accessible, clients need temporary credentials issued by the server in order to access it securely.
The reason do not implement this directly inside OSSFileIO is because we don’t want to limit to the client provided by the aliyun-sdk-oss, hope to the other filesystem implementations that can access OSS, which are configured via the fs.oss.impl property (such as Hadoop-Aliyun etc).

Reply:
@plusplusjiajia I see that we're setting the properties on the config, but I don't understand how this would work for S3A.  The hadoop implementation (S3A) uses a very different mechanism to apply credentials.  Also, there can be multiple credentials which is somewhat complicated due to Hadoop FileSystem caching by the uri authority.

I'm not convinced that this actually does anything.

Reply:
> @plusplusjiajia I see that we're setting the properties on the config, but I don't understand how this would work for S3A. The hadoop implementation (S3A) uses a very different mechanism to apply credentials. Also, there can be multiple credentials which is somewhat complicated due to Hadoop FileSystem caching by the uri authority.
> 
> I'm not convinced that this actually does anything.

Thanks for the reminder. It seems that need to handle the FileSystem cache, by default, storage credential and other configurations are not included in cache key.